### PR TITLE
Fix BOLD | ITALICS_A not getting escaped properly in MarkdownSanitizer (#1086)

### PR DESCRIPTION
**Pull request by V-ed**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: #1086

## Description

Fixes the `MarkdownSanitizer`'s `escape` method not properly escaping BOLD | ITALICS_A mixture of text (ie: `***text***`).

~~I did not test the changes, I will look how can I test later today.~~ 

~~I also plan to fix a very similar issue that I think might happen with code blocks, I will look into that later today too, making this PR a WIP one. I will therefore create a new commit, which I could squash them if preferable on any maintainer's request.~~ 

_Also, there might be a need to change the README's Contributing guidelines, as there is no `v4` branch anymore... :_ 
```
If you want to contribute to JDA, make sure to base your branch off of our v4 branch (or a feature-branch) and create your PR into that same branch. We will be rejecting any PRs between branches or into release branches!
```

### CODE REVIEW

**Comment by V-ed** [comment]
I will also look at adding / fixing tests later today, will update here accordingly.

**Comment by V-ed** [comment]
Hey @MinnDevelopment , I've got a question about your test suite - is your build step includes your unit test? I'm not used to gradle and I can't seem to make the tests fail, even when basically doing a assertTrue(false), and the build says it succeeded but one of the test should actually fail :

https://github.com/DV8FromTheWorld/JDA/blob/489ed4e222a937802d7f4341f0ac9ff1e9285f65/src/test/java/MarkdownTest.java#L283-L289

The first Assertion above should fail with my branch, as I fixed the escape function to be what is intended. I ran a manual test and indeed, this now works :
```java
public static void main(String[] args){
    String message = "***Test***";
    System.out.println(MarkdownSanitizer.escape(message));
}

>>  \*\*\*Test\*\*\*

**Comment by MinnDevelopment** [comment]
@V-ed I currently only run tests from intellij but they can be enabled in gradle with:

```kotlin
val test: Test by tasks

test.apply {
    useJUnitPlatform()
    failFast = true
}
```

Here is a patch you can apply:

```diff
diff --git a/build.gradle.kts b/build.gradle.kts
index bfcb0e85..df58c7aa 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -101,7 +101,7 @@ val javadoc: Javadoc by tasks
 val jar: Jar by tasks
 val build: Task by tasks
 val clean: Task by tasks
-val test: Task by tasks
+val test: Test by tasks
 val check: Task by tasks
 
 shadowJar.classifier = "withDependencies"
@@ -259,6 +259,11 @@ bintrayUpload.apply {
     onlyIf { System.getenv("BUILD_NUMBER") != null }
 }
 
+test.apply {
+    useJUnitPlatform()
+    failFast = true
+}
+
 publishing {
     publications {
         register("BintrayRelease", MavenPublication::class) {
```
You can apply the patch through `git apply` or with intellij **VCS** | **Apply Patch from Clipboard**

**Comment by V-ed** [comment]
After some testing, I concluded that there is no similar issues with code blocks (Discord formats the string `` \```text\``` `` as `` ```text``` `` correctly).

Althought, I knew there would be a problem with escaping every format as Discord's Markdown is a bit more convoluted than this, and indeed, the `monospace` function suffers from the lack of a proper escape mecanism :

image

as the second Assertion here would infer : https://github.com/DV8FromTheWorld/JDA/blob/489ed4e222a937802d7f4341f0ac9ff1e9285f65/src/test/java/MarkdownUtilTest.java#L48-L54

Should I also try to fix the monospace issue in this PR or would this be another scope, therefore another pull request?

**Comment by MinnDevelopment** [comment]
The monospace escape is working as intended, there is no reliable way of escaping it and this is an acceptable margin of error.

**Comment by V-ed** [comment]
> The monospace escape is working as intended, there is no reliable way of escaping it and this is an acceptable margin of error.

I actually beg to differ : there is a somewhat reliable way of escaping it, as I did in my own project, althought it's more of a workaround than a proper fix, and I do not believe that the image I've shown (and added below as well) is an acceptable margin of error, as it seriously degrades the actual format when formatting monospace with backticks in them.

image

As you can see below, that is what I would expect to see, which I believe *should* be the way to go, in expense of making the escaping mechanism a bit more complex :
image

... but I think this may be out of scope for this PR and that while there may be work on this later (at least, I hope), this PR here may be merged if accepted of course.

**Comment by MinnDevelopment** [comment]
> The monospace escape is working as intended

File: src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java [hunk 1]
```diff
@@ -416,6 +416,8 @@ private void applyStrategy(int region, @Nonnull String seq, @Nonnull StringBuild
             throw new IllegalStateException("Found illegal region for strategy ESCAPE '" + region + "' with no known format token!");
         if (region == UNDERLINE)
             token = "_\\_"; // UNDERLINE needs special handling because the client thinks its ITALICS_U if you only escape once
+        else if (region == (BOLD | ITALICS_A))
+            token = "*\\*\\*"; // BOLD | ITALICS_A needs special handling because the client thinks its BOLD if you only escape once
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java]: This probably also needs to be done for simple bold `\*\*` rather than `\**`

Comment (V-ed) [reply src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java]: Good catch, fixed - a few more tests were affected by this change, I tweaked them [in my latest commit](https://github.com/DV8FromTheWorld/JDA/pull/1087/commits/8f245bb7b3ef36f05c67fd67e7e4627ce7124560) to be more accurate on what's supposed to come out.

**Pull request merged**